x/dymns/keeper/hooks.go

```diff
@@ -0,0 +1,370 @@
+			)
+			// ignore the invalid entries for now, invariant will catch it
+			continue
+		}
+
+		if !so.HasFinished(blockEpochUTC) {
+			logger.Error(
+				"invalid entry on Active Sell Order Expiration records: Sell Order not yet finished.",
+				"asset-id", record.AssetId, "asset-type", assetType.PrettyName(),
+				"record-expiry", record.ExpireAt, "actual-expiry", so.ExpireAt,
+			)
+			// ignore the invalid entries for now, invariant will catch it
+			continue
+		}
+
+		finishedSellOrders = append(finishedSellOrders, *so)
+	}
+
+	return
+}
+
+/* -------------------------------------------------------------------------- */
+/*                             x/rollapp hooks                                */
+/* -------------------------------------------------------------------------- */
+
+// GetRollAppHooks returns the RollApp hooks struct.
+func (k Keeper) GetRollAppHooks() rollapptypes.RollappHooks {
+	return rollappHooks{
+		Keeper: k,
+	}
+}
+
+type rollappHooks struct {
+	Keeper
+}
+
+var _ rollapptypes.RollappHooks = rollappHooks{}
+
+func (h rollappHooks) RollappCreated(ctx sdk.Context, rollappID, alias string, creatorAddr sdk.AccAddress) error {
+	if alias == "" {
+		return nil
+	}
+
+	// ensure RollApp record is set
+	if !h.Keeper.IsRollAppId(ctx, rollappID) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "not a RollApp chain-id: %s", rollappID)
+	}
+
+	if !dymnsutils.IsValidAlias(alias) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "invalid alias format: %s", alias)
+	}
+
+	if !h.Keeper.CanUseAliasForNewRegistration(ctx, alias) {
+		return errorsmod.Wrapf(gerrc.ErrAlreadyExists, "alias already in use or preserved: %s", alias)
+	}
+
+	priceParams := h.Keeper.PriceParams(ctx)
+
+	aliasCost := sdk.NewCoins(
+		sdk.NewCoin(
+			priceParams.PriceDenom, priceParams.GetAliasPrice(alias),
+		),
+	)
+
+	return h.Keeper.registerAliasForRollApp(ctx, rollappID, creatorAddr, alias, aliasCost)
+}
+
+func (h rollappHooks) BeforeUpdateState(_ sdk.Context, _ string, _ string) error {
+	return nil
+}
+
+func (h rollappHooks) AfterStateFinalized(_ sdk.Context, _ string, _ *rollapptypes.StateInfo) error {
+	return nil
+}
+
+func (h rollappHooks) FraudSubmitted(_ sdk.Context, _ string, _ uint64, _ string) error {
+	return nil
+}
+
+// FutureRollappHooks is temporary added to handle future hooks that not available yet.
+type FutureRollappHooks interface {
+	OnRollAppIdChanged(ctx sdk.Context, previousRollAppId, newRollAppId string) error
+	// Just a pseudo method signature, the actual method signature might be different.
+
+	// TODO DymNS: connect to the actual implementation when the hooks are available.
+	//   The implementation of OnRollAppIdChanged assume that both of the RollApp records are exists in the x/rollapp store.
+}
+
+var _ FutureRollappHooks = rollappHooks{}
+
+func (k Keeper) GetFutureRollAppHooks() FutureRollappHooks {
+	return rollappHooks{
+		Keeper: k,
+	}
+}
+
```

Comment:
```suggestion
// OnRollAppIdChanged is called when a RollApp's ID is changed, typically due to fraud submission.
// It migrates all aliases and Dym-Names associated with the previous RollApp ID to the new one.
// This function executes in a branched context to prevent side effects, and any errors
// during execution will result in the state changes being discarded without returning an error.
//
// Parameters:
//   - ctx: The SDK context
//   - previousRollAppId: The original ID of the RollApp
//   - newRollAppId: The new ID assigned to the RollApp
//
// Returns:
//   - error: Always returns nil to prevent reverting critical executions in the caller
```

Reply:
if we never return an error here (always nil), what's the reason to have error as a return value?

Reply:
Because this is a hook, can be used by other places so better to give it an error in signature?

Reply:
not sure about it. cause even if we use it in other places, an err will never return (as we're not passing a flag for - should it return an err or not). imo we should remove.

Reply:
removed, thanks